src/entities/inputValidator/gyro/inputValidatorGyro.ts

```diff
@@ -6,15 +6,19 @@ import {
 import { PoolState } from '../../types';
 import { InputValidatorBase } from '../inputValidatorBase';
 import {
+    validateCreatePoolTokens,
     validateTokensAddLiquidity,
     validateTokensRemoveLiquidity,
 } from '../utils/validateTokens';
+import { validateCreatePoolTokenConfig } from '../utils/validateCreatePoolTokenConfig';
+
 import { inputValidationError, poolTypeError } from '@/utils';
 import { CreatePoolGyroECLPInput } from '@/entities/createPool';
 import { GyroECLPMath } from '@balancer-labs/balancer-maths';
 export class InputValidatorGyro extends InputValidatorBase {
     validateCreatePool(input: CreatePoolGyroECLPInput) {
-        super.validateCreatePool(input);
+        validateCreatePoolTokenConfig(input);
+        validateCreatePoolTokens(input.tokens);
```

Comment:
If I understood correctly, you moved these to each pool type specific implementations because you won't use this within the LBP implementation, right?
You could leave them in the `InputValidatorBase` and simply not call `super.validateCreatePool(input);` within the LBP implementation, which should achieve the same end result with less code duplication.
Should be ok to leave as is as well. Not a big deal.

Reply:
The main reason I moved them is because The `InputValidatorBase` was accessing the `tokens` of the `reatePoolInput` type. `tokens` however do not exist on the `CreatePoolLiquidityBootstrappingInput` type creating a mismatch. Therefor I had to move the create pool validation to each individual pool type validator.